### PR TITLE
[SPARK-36205][INFRA] Use set-env instead of set-output in GitHub Actions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR is more a cleanup. It removes unused `sync-branch` id in some steps, and use `set-env` instead of `set-output` to set an env.
This can be backported to branch-3.2 too.

### Why are the changes needed?

Cleanup.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

CI in this PR should test it out.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (Pending GitHub Action)

**Comment by dongjoon-hyun** [comment]
BTW, it would be great if we have a JIRA ID because it looks worth of backporting.

**Comment by HyukjinKwon** [comment]
Oh sure.

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -84,13 +84,12 @@ jobs:
...
     - name: Sync the current branch with the latest in Apache Spark
       if: github.repository != 'apache/spark'
-      id: sync-branch
       run: |
         apache_spark_ref=`git rev-parse HEAD`
         git fetch https://github.com/$GITHUB_REPOSITORY.git ${GITHUB_REF#refs/heads/}
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' merge --no-commit --progress --squash FETCH_HEAD
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' commit -m "Merged commit"
-        echo "::set-output name=APACHE_SPARK_REF::$apache_spark_ref"
+        echo "::set-env name=APACHE_SPARK_REF::$apache_spark_ref"
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:92]:
```suggestion
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -182,13 +180,12 @@ jobs:
...
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:188]:
```suggestion
```

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -182,13 +180,12 @@ jobs:
         ref: master
     - name: Sync the current branch with the latest in Apache Spark
       if: github.repository != 'apache/spark'
-      id: sync-branch
       run: |
         apache_spark_ref=`git rev-parse HEAD`
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:184]:
```suggestion
        echo "APACHE_SPARK_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
```

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -84,13 +84,12 @@ jobs:
...
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:88]:
```suggestion
        echo "APACHE_SPARK_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
```

File: .github/workflows/build_and_test.yml [hunk 5]
```diff
@@ -254,13 +250,12 @@ jobs:
...
         git fetch https://github.com/$GITHUB_REPOSITORY.git ${GITHUB_REF#refs/heads/}
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' merge --no-commit --progress --squash FETCH_HEAD
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' commit -m "Merged commit"
-        echo "::set-output name=APACHE_SPARK_REF::$apache_spark_ref"
+        echo "::set-env name=APACHE_SPARK_REF::$apache_spark_ref"
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:258]:
```suggestion
```

File: .github/workflows/build_and_test.yml [hunk 6]
```diff
@@ -254,13 +250,12 @@ jobs:
         ref: master
     - name: Sync the current branch with the latest in Apache Spark
       if: github.repository != 'apache/spark'
-      id: sync-branch
       run: |
         apache_spark_ref=`git rev-parse HEAD`
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:254]:
```suggestion
        echo "APACHE_SPARK_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
```

File: .github/workflows/build_and_test.yml [hunk 7]
```diff
@@ -632,13 +621,12 @@ jobs:
...
         git fetch https://github.com/$GITHUB_REPOSITORY.git ${GITHUB_REF#refs/heads/}
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' merge --no-commit --progress --squash FETCH_HEAD
         git -c user.name='Apache Spark Test Account' -c user.email='sparktestacc@gmail.com' commit -m "Merged commit"
-        echo "::set-output name=APACHE_SPARK_REF::$apache_spark_ref"
+        echo "::set-env name=APACHE_SPARK_REF::$apache_spark_ref"
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:629]:
```suggestion
```

File: .github/workflows/build_and_test.yml [hunk 8]
```diff
@@ -632,13 +621,12 @@ jobs:
         ref: master
     - name: Sync the current branch with the latest in Apache Spark
       if: github.repository != 'apache/spark'
-      id: sync-branch
       run: |
         apache_spark_ref=`git rev-parse HEAD`
```
Comment (HyukjinKwon) [inline .github/workflows/build_and_test.yml:625]:
```suggestion
        echo "APACHE_SPARK_REF=$(git rev-parse HEAD)" >> $GITHUB_ENV
```

**Comment by HyukjinKwon** [comment]
all tests passed. I pushed another commit to check if the env is really set for doubly sure. I will revert the last commit and merge once that's verified.

**Comment by HyukjinKwon** [comment]
okay it does correctly:
Screen Shot 2021-07-19 at 7 41 05 PM
Screen Shot 2021-07-19 at 7 41 28 PM

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.2

**Pull request closed without merging**